addons/resource/models/resource.py

```diff
@@ -536,193 +481,176 @@ def _schedule_days(self, cr, uid, id, days, day_date=None, compute_leaves=False,
         planned_days = 0
         iterations = 0
         current_datetime = day_date.replace(hour=0, minute=0, second=0)
-
         while planned_days < days and iterations < 100:
             working_intervals = self.get_working_intervals_of_day(
-                cr, uid, id, current_datetime,
+                current_datetime,
                 compute_leaves=compute_leaves, resource_id=resource_id,
-                default_interval=default_interval,
-                context=context)
-            if id is None or working_intervals:  # no calendar -> no working hours, but day is considered as worked
+                default_interval=default_interval)
+            if not self or working_intervals:  # no calendar -> no working hours, but day is considered as worked
                 planned_days += 1
                 intervals += working_intervals
             # get next day
             if backwards:
-                current_datetime = self.get_previous_day(cr, uid, id, current_datetime, context)
+                current_datetime = self.get_previous_day(current_datetime)
             else:
-                current_datetime = self.get_next_day(cr, uid, id, current_datetime, context)
+                current_datetime = self.get_next_day(current_datetime)
             # avoid infinite loops
             iterations += 1
-
         return intervals
 
-    def schedule_days_get_date(self, cr, uid, id, days, day_date=None, compute_leaves=False,
-                               resource_id=None, default_interval=None, context=None):
+    @api.multi
+    def schedule_days_get_date(self, days, day_date=None, compute_leaves=False,
+                               resource_id=None, default_interval=None):
         """ Wrapper on _schedule_days: return the beginning/ending datetime of
         a days scheduling. """
-        res = self._schedule_days(cr, uid, id, days, day_date, compute_leaves, resource_id, default_interval, context)
+        res = self._schedule_days(days, day_date, compute_leaves, resource_id, default_interval)
         return res and res[-1][1] or False
 
-    def schedule_days(self, cr, uid, id, days, day_date=None, compute_leaves=False,
-                      resource_id=None, default_interval=None, context=None):
+    def schedule_days(self, days, day_date=None, compute_leaves=False,
+                      resource_id=None, default_interval=None):
         """ Wrapper on _schedule_days: return the working intervals of a days
         scheduling. """
-        return self._schedule_days(cr, uid, id, days, day_date, compute_leaves, resource_id, default_interval, context)
+        return self._schedule_days(days, day_date, compute_leaves, resource_id, default_interval)
 
     # --------------------------------------------------
     # Compatibility / to clean / to remove
     # --------------------------------------------------
-
+    @api.v7
     def working_hours_on_day(self, cr, uid, resource_calendar_id, day, context=None):
```

Comment:
give parameter name "resource_calendar" instead of "resource_calendar_id" . because we got recordset(s) as argument so easily identify based on name.